### ISSUE TITLE
[F#] fatal exception when building a solution

### ISSUE THREAD

**Issue by nosami** [mono/monodevelop#5049]
```
[ERROR] FATAL UNHANDLED EXCEPTION: Microsoft.FSharp.Compiler.ErrorLogger+ReportedError: The exception has been reported. This internal exception should now be caught at an error recovery point on the stack. Original message: The file '/Users/jason/src/Elmish.XamarinForms/packages/macos/Xamarin.Forms/lib/Xamarin.Mac/Xamarin.Forms.Platform.macOS.dll' changed on disk unexpectedly, please reload.)
  at Microsoft.FSharp.Compiler.ErrorLogger+ErrorLoggerExtensions.ErrorLogger.Error[b] (Microsoft.FSharp.Compiler.ErrorLogger+ErrorLogger x, System.Exception exn) [0x00013] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Compiler.AbstractIL.ILBinaryReader+WeakByteFile.Microsoft-FSharp-Compiler-AbstractIL-ILBinaryReader-BinaryFile-GetView () [0x00054] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Compiler.AbstractIL.ILBinaryReader.seekReadCustomAttributeRow (Microsoft.FSharp.Compiler.AbstractIL.ILBinaryReader+ILMetadataReader ctxt, System.Int32 idx) [0x00000] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Compiler.AbstractIL.ILBinaryReader+customAttrsReader@2605-1.Invoke (System.Int32 idx) [0x00000] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Compiler.AbstractIL.ILBinaryReader.seekReadIndexedRows$cont@971[a,b,c] (Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] rowReader, Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] rowConverter, System.Int32 numRows, Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] keyFunc, Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] keyComparer, Microsoft.FSharp.Core.Unit unitVar) [0x0001c] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Compiler.AbstractIL.ILBinaryReader.seekReadIndexedRows[a,b,c] (System.Int32 numRows, Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] rowReader, Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] keyFunc, Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] keyComparer, System.Boolean binaryChop, Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] rowConverter) [0x00004] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Compiler.AbstractIL.ILBinaryReader+customAttrsReader@2602.Invoke (System.Int32 idx) [0x0004b] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Compiler.AbstractIL.IL+ILAttributesStored.GetCustomAttrs (System.Int32 metadataIndex) [0x00022] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Compiler.AbstractIL.IL+ILTypeDef.get_CustomAttrs () [0x00000] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Compiler.AttributeChecking.GetAttribInfosOfEntity (Microsoft.FSharp.Compiler.TcGlobals+TcGlobals g, Microsoft.FSharp.Compiler.Import+ImportMap amap, Microsoft.FSharp.Compiler.Range+range m, Microsoft.FSharp.Compiler.Tast+EntityRef tcref) [0x00037] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Compiler.SourceCodeServices.FSharpEntity.get_Attributes () [0x0003b] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Compiler.SourceCodeServices.UnusedOpens+symbols@23-1.GenerateNext (System.Collections.Generic.IEnumerable`1[Microsoft.FSharp.Compiler.SourceCodeServices.FSharpSymbol]& next) [0x00220] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1[T].MoveNextImpl () [0x00016] in <5a7203e1904cf4daa7450383e103725a>:0
  at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1[T].System-Collections-IEnumerator-MoveNext () [0x00000] in <5a7203e1904cf4daa7450383e103725a>:0
  at System.Collections.Generic.List`1[T].AddEnumerable (System.Collections.Generic.IEnumerable`1[T] enumerable) [0x00059] in /Users/builder/jenkins/workspace/build-package-osx-mono/master/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/List.cs:1109
  at System.Collections.Generic.List`1[T]..ctor (System.Collections.Generic.IEnumerable`1[T] collection) [0x00062] in /Users/builder/jenkins/workspace/build-package-osx-mono/master/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/List.cs:88
  at Microsoft.FSharp.Collections.SeqModule.ToArray[T] (System.Collections.Generic.IEnumerable`1[T] source) [0x00069] in <5a7203e1904cf4daa7450383e103725a>:0
  at Microsoft.FSharp.Compiler.SourceCodeServices.UnusedOpens+-ctor@21-147.Invoke (Microsoft.FSharp.Core.Unit unitVar) [0x00019] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at System.Lazy`1[T].CreateValue () [0x00081] in /Users/builder/jenkins/workspace/build-package-osx-mono/master/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/Lazy.cs:443
  at System.Lazy`1[T].LazyInitValue () [0x00080] in /Users/builder/jenkins/workspace/build-package-osx-mono/master/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/Lazy.cs:388
  at System.Lazy`1[T].get_Value () [0x0003a] in /Users/builder/jenkins/workspace/build-package-osx-mono/master/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/Lazy.cs:339
  at Microsoft.FSharp.Compiler.SourceCodeServices.UnusedOpens+getUsedModules@166-5.Invoke (Microsoft.FSharp.Compiler.SourceCodeServices.FSharpSymbolUse symbolUse) [0x0001f] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Collections.ArrayModule.loop@370-34[T] (Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] predicate, T[] array, System.Int32 len, System.Int32 i) [0x00004] in <5a7203e1904cf4daa7450383e103725a>:0
  at Microsoft.FSharp.Collections.ArrayModule.Exists[T] (Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] predicate, T[] array) [0x00024] in <5a7203e1904cf4daa7450383e103725a>:0
  at Microsoft.FSharp.Primitives.Basics.List.exists[T] (Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] predicate, Microsoft.FSharp.Collections.FSharpList`1[T] xs1) [0x00018] in <5a7203e1904cf4daa7450383e103725a>:0
  at Microsoft.FSharp.Collections.ListModule.Exists[T] (Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] predicate, Microsoft.FSharp.Collections.FSharpList`1[T] list) [0x00000] in <5a7203e1904cf4daa7450383e103725a>:0
  at Microsoft.FSharp.Primitives.Basics.List.filter[T] (Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] predicate, Microsoft.FSharp.Collections.FSharpList`1[T] l) [0x00040] in <5a7203e1904cf4daa7450383e103725a>:0
  at Microsoft.FSharp.Collections.ListModule.Filter[T] (Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] predicate, Microsoft.FSharp.Collections.FSharpList`1[T] list) [0x00000] in <5a7203e1904cf4daa7450383e103725a>:0
  at Microsoft.FSharp.Compiler.SourceCodeServices.UnusedOpens.getUsedModules (Microsoft.FSharp.Compiler.SourceCodeServices.FSharpSymbolUse[] symbolUses1, Microsoft.FSharp.Compiler.SourceCodeServices.FSharpSymbolUse[] symbolUses2, Microsoft.FSharp.Collections.FSharpList`1[T] usedModules, Microsoft.FSharp.Compiler.SourceCodeServices.UnusedOpens+OpenStatement openStatement) [0x00021] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Compiler.SourceCodeServices.UnusedOpens+filterOpenStatementsIncremental@178.Invoke (Microsoft.FSharp.Core.Unit unitVar) [0x0003a] in <5b0f50bc36f83639a7450383bc500f5b>:0
  at Microsoft.FSharp.Control.AsyncBuilderImpl+callA@522[b,a].Invoke (Microsoft.FSharp.Control.AsyncParams`1[T] args) [0x00051] in <5a7203e1904cf4daa7450383e103725a>:0
```

> VS bug [#632894](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/632894)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.342 (master)

Pull Request #5174 merged by: Jason Imison
Author: mono
Commit: 305bddc1e9dcd90caf25f8bed1a042efc317528b (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #5174 merged by: Jason Imison
Author: mono
Commit: 305bddc1e9dcd90caf25f8bed1a042efc317528b (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-17] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #5174 merged by: Jason Imison  
 Author: mono  
 Commit: 305bddc1e9dcd90caf25f8bed1a042efc317528b (xamarin/monodevelop)
